src/test_rig_bluesky/plans.py

```diff
@@ -29,9 +31,12 @@ def spectroscopy(
     oav: AravisDetector = oav,
     sample_stage: XYZStage = sample_stage,
     spec: Spec | None = None,
+    exposure_time: float = 0.1,
     metadata: dict[str, Any] | None = None,
 ) -> MsgGenerator[None]:
     """Do a spectroscopy scan."""
+    yield from bps.prepare(oav, TriggerInfo(livetime=exposure_time))
```

Comment:
Is this the correct way to set exposure?

Reply:
The code path here is a tad complicated so maybe worth explaining. This ultimately leads to [this line of code](https://github.com/bluesky/ophyd-async/blob/864c2c2574750a9405fd695a1b217a25ded6557f/src/ophyd_async/epics/adaravis/_aravis_controller.py#L22) getting called, which does what you want.

Later in the scan the `RunEngine` will call [`trigger()`](https://github.com/bluesky/ophyd-async/blob/864c2c2574750a9405fd695a1b217a25ded6557f/src/ophyd_async/core/_detector.py#L274) on the detector, which calls prepare again but with `livetime=None`, which will leave your exposure time unchanged.

So yes, this is right, but `abs_set`/`mv` are also valid ways of doing this.